sampler.go

```diff
@@ -199,45 +200,74 @@ func (s *rateLimitingSampler) Equal(other Sampler) bool {
 // The probabilisticSampler is given higher priority when tags are emitted, ie. if IsSampled() for both
 // samplers return true, the tags for probabilisticSampler will be used.
 type GuaranteedThroughputProbabilisticSampler struct {
-	probabilisticSampler Sampler
-	lowerBoundSampler    Sampler
-	tags                 []Tag
-	samplingRate         float64
-	lowerBound           float64
+	probabilisticSampler  Sampler
+	lowerBoundSampler     Sampler
+	upperBoundRateLimiter utils.RateLimiter
+	tags                  []Tag
+	samplingRate          float64
+	lowerBound            float64
+	upperBound            float64
 }
 
 // NewGuaranteedThroughputProbabilisticSampler returns a delegating sampler that applies both
 // probabilisticSampler and rateLimitingSampler.
 func NewGuaranteedThroughputProbabilisticSampler(
-	lowerBound, samplingRate float64,
+	lowerBound, samplingRate, upperBound float64,
 ) (*GuaranteedThroughputProbabilisticSampler, error) {
-	tags := []Tag{
-		{key: SamplerTypeTagKey, value: SamplerTypeLowerBound},
-		{key: SamplerParamTagKey, value: samplingRate},
-	}
 	probabilisticSampler, err := NewProbabilisticSampler(samplingRate)
 	if err != nil {
```

Comment:
I'd love to do this but if we change the constructor to not return an error, it would become a breaking change. For now, I'll make this change but keep the API as is and always swallow the error whenever we call NewProbabilisticSampler and add a TODO and an issue to remove the error from the API when we do a major upgrade.